components/tidb_query/src/expr/mod.rs

```diff
@@ -297,7 +295,6 @@ impl Expression {
                             sig: expr.get_sig(),
                             children,
                             field_type,
-                            implicit_args,
```

Comment:
I think you need to at least pass the `val()` inside. Otherwise when you extract constant arguments to metadata in `IN()`, it's non-vectorized implementation no longer works, which will break the compatibility.